### PR TITLE
Render the console inside the body tag

### PR DESCRIPTION
**Pull request by gsamokovarov**
Before this commit, the console was rendered straight after the closing
</html> tag, which made the page HTML invalid. However, browsers are
used to be molested and they are really good at displaying whatever you
throw at them, so that wasn't big problem anyway. Still it was something
that bug me quite a bit. I'm trying to redeem myself here.

Because I had to change the newly introduced `WebConsole::Response` class
I had cleaned the workspace a bit:

* `WebConsole::Response` no longer depends on a logger. The detection
  should be better now, so we don't need to couple the query check in
  it.

* `WebConsole::Response` does one thing only. It used to check the
  acceptance of a given `Content-Type`. I moved this to the middleware
  class.

* I have reverted the `WebConsole::WhinyRequest` to keep the logging out
  of the logic in the request class. We no longer need to log in two
  separate classes, so no need to break `Liskov` or introduce logging
  into simple query methods.

@sh19910711 do you wanna review this?

### CODE REVIEW

File: lib/web_console/response.rb [hunk 1]
```diff
@@ -1,37 +1,26 @@
...
-      @logger = logger
-      super(body, status, headers)
-    end
-
+  # A response object that writes content before the closing </body> tag, if
+  # possible.
+  #
+  # The object quacks like Rack::Response.
+  class Response < Struct.new(:body, :status, :headers)
     # Returns whether the response is from an acceptable content type.
```
Comment (sh19910711) [inline lib/web_console/response.rb]: Need to remove this comment?

**Comment by sh19910711** [comment]
> @sh19910711 do you wanna review this?

Yeah, I will do.

File: lib/web_console/response.rb (hunk 1 above)
Comment (gsamokovarov) [inline lib/web_console/response.rb]: :+1:

**Comment by gsamokovarov** [comment]
@sh19910711 I've updated the PR.

File: test/web_console/middleware_test.rb [hunk 2]
```diff
@@ -50,6 +54,26 @@ def headers
       assert_select '#console'
     end
 
+    test 'render console inside the body tag' do
+      get '/', nil, 'web_console.binding' => binding
+
+      assert_select 'body > #console'
+    end
+
+    test '#acceptable_content_type? is truthy if response format is HTML' do
```
Comment (sh19910711) [inline test/web_console/middleware_test.rb]: This test case seems to be doing the same thing as the above one (`test 'render console inside the body tag'`). We can merge them into one, perhaps.

File: test/web_console/middleware_test.rb [hunk 3]
```diff
@@ -50,6 +54,26 @@ def headers
...
+      @app = Middleware.new(Application.new(response_content_type: Mime::HTML))
+      get '/', nil, 'web_console.binding' => binding
+
+      assert_select '#console'
+    end
+
+    test '#acceptable_content_type? is falsy if response format is not HTML' do
```
Comment (sh19910711) [inline test/web_console/middleware_test.rb]:
```diff
-    test '#acceptable_content_type? is falsy if response format is not HTML' do
+    test 'does not render console if response format is not HTML' do
```

or something else fitting.

**Comment by sh19910711** [comment]
The code seems good to me.

> WebConsole::Response does one thing only.

I like that :+1:

File: test/web_console/middleware_test.rb (hunk 3 above)
Comment (gsamokovarov) [inline test/web_console/middleware_test.rb]: Yes, good catch.

File: test/web_console/middleware_test.rb (hunk 2 above)
Comment (gsamokovarov) [inline test/web_console/middleware_test.rb]: Yup.

**Pull request merged**